package.json

```diff
@@ -24,60 +24,61 @@
+    "test": "mocha -r @babel/register -R spec ./test/unit/**",
     "test:cov": "nyc --reporter=lcov --reporter=html npm run test",
     "codecov": "cat coverage/lcov.info | codecov",
     "lint": "eslint src test",
     "lint:fix": "npm run lint -- --fix",
     "format": "prettier --write \"src/**/*.js\" \"test/**/*.js\"",
     "prepare": "npm run clean && npm run build",
-    "preversion": "npm run clean && npm run build",
+    "prepublish": "npm run clean && npm run build",
     "pre-push": "npm run lint"
   },
   "dependencies": {
-    "immer": "8.0.1",
-    "lodash": "^4.17.20",
+    "debug": "^4.3.1",
+    "immer": "9.0.1",
+    "lodash": "^4.17.21",
     "reduce-reducers": "^1.0.4"
   },
   "devDependencies": {
-    "@babel/cli": "^7.12.16",
-    "@babel/core": "^7.12.16",
-    "@babel/plugin-transform-modules-commonjs": "^7.12.13",
-    "@babel/plugin-transform-runtime": "^7.12.15",
-    "@babel/preset-env": "^7.12.16",
-    "@babel/preset-react": "^7.12.13",
-    "@babel/register": "^7.12.13",
-    "@babel/runtime": "^7.12.13",
+    "@babel/cli": "^7.13.14",
+    "@babel/core": "^7.13.15",
+    "@babel/plugin-transform-modules-commonjs": "^7.13.8",
+    "@babel/plugin-transform-runtime": "^7.13.15",
+    "@babel/preset-env": "^7.13.15",
+    "@babel/preset-react": "^7.13.13",
+    "@babel/register": "^7.13.14",
+    "@babel/runtime": "^7.13.10",
     "babel-eslint": "^10.1.0",
     "babel-loader": "^8.2.2",
     "babel-plugin-lodash": "^3.3.4",
     "babel-plugin-module-resolver": "^4.1.0",
     "babel-plugin-transform-inline-environment-variables": "^0.4.3",
-    "chai": "^4.3.0",
+    "chai": "^4.3.4",
     "codecov": "^3.8.1",
     "cross-env": "^7.0.3",
-    "eslint": "^6.8.0",
+    "eslint": "^7.24.0",
     "eslint-config-airbnb-base": "^14.2.1",
     "eslint-config-google": "^0.14.0",
-    "eslint-config-prettier": "^6.10.0",
-    "eslint-plugin-babel": "^5.3.0",
-    "eslint-plugin-import": "^2.20.1",
-    "eslint-plugin-jsdoc": "^21.0.0",
-    "eslint-plugin-jsx-a11y": "^6.2.3",
-    "eslint-plugin-prettier": "^3.1.0",
-    "eslint-plugin-react": "^7.18.3",
-    "firebase": "^7.8.0",
-    "husky": "^4.2.1",
+    "eslint-config-prettier": "^8.2.0",
+    "eslint-plugin-babel": "^5.3.1",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsdoc": "^32.3.0",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.4.0",
+    "eslint-plugin-react": "^7.23.2",
+    "firebase": "^8.4.1",
+    "husky": "^6.0.0",
     "mocha": "^7.0.1",
-    "nyc": "^15.0.0",
-    "prettier": "1.19.1",
-    "redux": "^4.0.1",
-    "rimraf": "^3.0.0",
-    "sinon": "^8.1.1",
-    "sinon-chai": "^3.4.0",
-    "uglifyjs-webpack-plugin": "^2.2.0",
-    "webpack": "^4.41.5",
-    "webpack-cli": "^3.3.10"
+    "nyc": "^15.1.0",
+    "prettier": "2.2.1",
+    "redux": "^4.0.5",
+    "rimraf": "^3.0.2",
+    "sinon": "^10.0.0",
+    "sinon-chai": "^3.6.0",
+    "terser-webpack-plugin": "^5.1.1",
+    "webpack": "^5.33.2",
```

Comment:
Really nice to see these things updated, but will have to check the difference in webpack 5 output vs 4 to make sure there isn't anything breaking happening

Were any of these dependency updates needed for the functionality you were building or did you just capture it all together? If not, it might be good to get these depedency updates out in a separate PR while we are still discussing the immediate mode API/implementation

Reply:
Recalling this out here - we should be able to add the cache reducer without these updates right?